### PR TITLE
Revert "[Triggers] disable tests flaky on windows (#15955)"

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
This reverts commit 8201064bbf6de00a3c9d6e6da2b368b4b3037713.

- [x] Factored failure testing into its own file (i.e. FailureTesting.scala) along with a separate build/test target
- [x] Added in comments to ensure that FailureTesting.scala and Cats.daml have `contractPairings` values synchronised
- [x] Added in a PP restriction to trigger logging of api.Value message lists

### CODE REVIEW

File: triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/test/AbstractFuncTests.scala [hunk 1]
```diff
@@ -41,7 +41,7 @@ abstract class AbstractFuncTests
 
   this.getClass.getSimpleName can {
     "Failure testing" should {
-      val contractPairings = 500
+      val contractPairings = 400
```
Comment (carlpulley-da) [inline triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/test/AbstractFuncTests.scala:44]: Looking at failing Windows test logging, I believe I'm observing only ~940 contracts being created on windows systems within a 5 minute test window. Reducing the number of contract pairings to 400 should hopefully allow these tests to run (without being flaky) in both Windows and *nix environments?

File: triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Runner.scala [hunk 2]
```diff
@@ -1313,7 +1313,7 @@ object Runner {
         )
 
     implicit def `api.Value to LoggingValue`: ToLoggingValue[api.Value] = value =>
-      PrettyPrint.prettyApiValue(verbose = true)(value).render(80)
+      PrettyPrint.prettyApiValue(verbose = true, maxListWidth = Some(20))(value).render(80)
```
Comment (carlpulley-da) [inline triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Runner.scala:1316]: Restrict PP of message lists in logging

File: triggers/tests/BUILD.bazel [hunk 3]
```diff
@@ -126,6 +126,7 @@ da_scala_library(
 [
     da_scala_test_suite(
         name = "trigger-integration-tests" + suffix,
+        size = "large",
```
[the remark this answers is not in the document]
Comment (carlpulley-da) [reply triggers/tests/BUILD.bazel:129]: Understood. This change was purely for testing purposes.

File: triggers/tests/BUILD.bazel [hunk 4]
```diff
@@ -184,3 +183,56 @@ da_scala_library(
     for lf_version in DAML_LF_VERSIONS
     for suffix in [("-" + lf_version) if lf_version else ""]
 ]
+
+[
+    da_scala_test_suite(
+        name = "trigger-failure-integration-tests" + suffix,
+        srcs = [
+            "src/test/scala/com/digitalasset/daml/lf/engine/trigger/test/FailureTesting.scala",
```
Comment (carlpulley-da) [inline triggers/tests/BUILD.bazel:191]: Separate testing of FailureTesting.scala for greater build/CI control of this expensive test

**Review by remyhaemmerle-da — COMMENTED** [review]
LGTM. Thanks.

**Review by carlpulley-da — APPROVED** [review]

**Pull request merged**